pkgs/development/octave-modules/symbolic/default.nix

```diff
@@ -0,0 +1,59 @@
+{ buildOctaveLibrary
+, stdenv
+, fetchurl
+, octave
+, gnuplot
+, python
+, python2Packages
+, texinfo
+}:
+
+let
+  # Need to use sympy 1.5.1 for https://github.com/cbm755/octsympy/issues/1023
+  # It has been addressed, but not merged yet.
+  # In the meantime, we create a Python environment with Python 3, its mpmath
+  # version and sympy 1.5 from python2Packages.
+  pythonEnv = (let
+      overridenPython = let
+        packageOverrides = self: super: {
+          sympy = super.sympy.overridePythonAttrs (old: rec {
+            version = python2Packages.sympy.version;
+            src = python2Packages.sympy.src;
+          });
+        };
+      in python.override {inherit packageOverrides; self = overridenPython; };
+    in overridenPython.withPackages (ps: [
+      ps.sympy
+      ps.mpmath
+    ]));
+
+in (buildOctaveLibrary rec {
+  pname = "symbolic";
+  version = "2.9.0";
+
+  src = fetchurl {
+    url = "https://octave.sourceforge.io/download.php?package=${pname}-${version}.tar.gz";
+    sha256 = "1jr3kg9q6r4r4h3hiwq9fli6wsns73rqfzkrg25plha9195c97h8";
+  };
+
+  buildInputs = [ pythonEnv ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://octave.sourceforge.io/${pname}/index.html";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ KarlJoad ];
+    description = "Adds symbolic calculation features to GNU Octave";
+  };
+}).overrideAttrs (oldAttrs: rec {
+  dontUnpack = false;
+  buildPhase = ''
+      substituteInPlace inst/private/defaultpython.m --replace python3 ${pythonEnv}/bin/python
```

Comment:
A ~year ago, I had a working octave with sympy, where the only two things I did were:

1. Wrapping octave with `--prefix PATH ${pythonEnv}/bin` and pythonEnv was similar to yours.
2. Adding the package with `pkg add_prefix` and `pkg load symbolic`.

That worked (until symbolic broke due to the new sympy). Maybe we should handle packages like symbolic differently? Perhaps by introducing a hook?

Reply:
Right now, I have managed to find a way to not need the override at all. Instead of re-writing the python interpreter (that has an environment where sympy and mpmath are present), that particular python has been added to the `PATH`, so that octave can perform a normal look-up of `python3` in the environment variables.

In theory, this should work for any other package that also uses `PATH` to find the location of its other run-time dependencies.